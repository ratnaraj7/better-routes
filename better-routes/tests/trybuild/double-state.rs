use axum::response::{IntoResponse, Response};
use axum_extra::routing::RouterExt;
use better_routes::{method_helper, routes};

#[derive(Default)]
struct GlobalRejection;

impl IntoResponse for GlobalRejection {
    fn into_response(self) -> Response {
        todo!()
    }
}

#[derive(Default)]
struct FooRejection;

impl IntoResponse for FooRejection {
    fn into_response(self) -> Response {
        todo!()
    }
}

#[derive(Clone)]
struct StateOne {}

#[derive(Clone)]
struct StateTwo {}

routes! {
    State => StateOne,
    State => StateTwo,
    Rejection => GlobalRejection,
    "/foo" => struct Foo; => FooRejection,
    "/bar" => struct Bar;,
}

#[method_helper(StateOne)]
impl Foo {
    #[get]
    async fn foo(self) {}
}

#[method_helper(StateTwo)]
impl Bar {
    #[get]
    async fn bar(self) {}
}

fn main() {}
