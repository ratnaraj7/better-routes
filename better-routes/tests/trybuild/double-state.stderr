error: Duplicate State
  --> tests/trybuild/double-state.rs:31:5
   |
31 |     State => StateTwo,
   |     ^^^^^

error[E0412]: cannot find type `Foo` in this scope
  --> tests/trybuild/double-state.rs:38:6
   |
38 | impl Foo {
   |      ^^^ not found in this scope

error[E0412]: cannot find type `Bar` in this scope
  --> tests/trybuild/double-state.rs:44:6
   |
44 | impl Bar {
   |      ^^^ not found in this scope
